common/djangoapps/enrollment/views.py

```diff
@@ -469,6 +472,14 @@ def get(self, request):
                     ).format(username=username)
                 }
             )
+        if username == request.user.username or self.has_api_key_permissions(request):
+            return Response(enrollment_data)
+        filtered_data = []
+        for enrollment in enrollment_data:
```

Comment:
That makes sense.  Perhaps a product perspective would clear things up then.  @explorerleslie (I think your'e the PO for this work), it looks like organizational staff and instructors wouldn't have access.